### ISSUE TITLE
Erase barebox environment 

### ISSUE THREAD

**Issue by vishwanathchandapur** [rauc/rauc#418]
We have board with emmc and barebox  bootloader, We are trying to experiment with ruac barebox update .while testing we found rauc doesn't clear barebox environment after upgrade because which next boot old barebox environment will be used.

**Comment by jluebbe** [comment]
I'd recommend avoiding a barebox environment for any non-development use-case. You can store dynamic data with a fixed schema using the state framework: https://barebox.org/doc/latest/user/state.html.

**Comment by ejoerns** [comment]
@vishwanathchandapur there is nothing wrong with that, that is what one would expect. The environment should be persistent as it might contain changes unrelated to the specific barebox you use.

But this should not be an issue as you should not have anything in you barebox env on a production system. The barebox env is only meant for developing or debugging purpose!

When you update barebox on eMMC, do you use the boot partitions for that?

**Comment by ejoerns** [comment]
Ah, that happens if no reloading the browser before finally submitting a pre-written answer... :D

**Comment by vishwanathchandapur** [comment]
Yes as of now  we are using boot partitions and we trying experiment  with barebox update on filed if this doesn't have any side effects .
Coming to your solution in scenarios where if try to pass new argument in linux boot arguments ,the new changes wont be reflected untill we clear barebox environment?

**Comment by ejoerns** [comment]
If using barebox I would recommend using [bootspec](https://www.freedesktop.org/wiki/Specifications/BootLoaderSpec/) files for adding linux boot arguments so that they are part of the rootfs instead of placing them in the barebox env.

> Yes as of now we are using boot partitions and we trying experiment with barebox update on filed if doesn't have any side effects .

I did not get what you actually mean by barebox update on file{d,s}. Could you rephrase that?

**Comment by ejoerns** [comment]
@vishwanathchandapur Any updates on this? Does it work for you now?

**Comment by ejoerns** [comment]
I will close this now as there was no feedback for 5 months and there does not seem to be any general issue here.
